### ISSUE TITLE
Checkbox bugs

### ISSUE THREAD

**Issue by Jip-Hop** [galilley/syncthing-pyselective#2]
Great to see you're building a solution for convenient selective sync! I've been trying it out but run into some issues.

I applied the ignore template to my test folder, to prepare it for syncthing-pyselective:
```
//* Do not edit comments like this! *//

//* Global ignores *//

!/.stignoreglobal

// Uncomment the next file if only the file exists locally

//#include .stignoreglobal

//* User defined *//

// optional local ignores

//* Selective sync (generated by pyselective) *//

!/folder

//* ignore all except selected *//

**
```

Then I run syncthing-pyselective. All folders are ignored by default, which is expected.

<img width="516" alt="Screenshot 2021-07-05 at 16 52 47" src="https://user-images.githubusercontent.com/2871973/124489935-d193b400-ddb1-11eb-9f6e-4aa137cb6962.png">

Then I check a folder.

<img width="520" alt="Screenshot 2021-07-05 at 16 53 03" src="https://user-images.githubusercontent.com/2871973/124489957-d7899500-ddb1-11eb-85f2-d4265c92b31f.png">

But when I expand the folder, the 'checked' state disappears. Also, the items inside the folder are yellow (globalignore). No idea why as I don't have any custom ignore rules.

<img width="517" alt="Screenshot 2021-07-05 at 16 53 13" src="https://user-images.githubusercontent.com/2871973/124490009-e8d2a180-ddb1-11eb-82da-62f02a0ecfb7.png">

Re-checking the folder I want to sync still doesn't check the sub-items and I'm not allowed to check any of the yellow items.

<img width="515" alt="Screenshot 2021-07-05 at 16 53 21" src="https://user-images.githubusercontent.com/2871973/124490258-2afbe300-ddb2-11eb-8645-f40484331820.png">

Expanding the yellow folder and checking a sub-item can lead to inconsistent checkbox state (unchecked box for firefox-87.0.source, even though some children are checked).

<img width="833" alt="Screenshot 2021-07-05 at 17 04 02" src="https://user-images.githubusercontent.com/2871973/124491034-19ffa180-ddb3-11eb-8c97-62a54511a3fe.png">

How is syncthing-pyselective working out for you? I hope you continue to improve it 🙂 

P.S.
Tested with syncthing-pyselective-0.2.1 on MacOS.

**Comment by galilley** [comment]
Dear Jip-Hop, 
thank you  for so detailed description!

In fact work with checkboxes is a quite tricky. It this moment I'm tracking a state of the parent checkbox but do not propagate it for children. That is why it clears while expanding. I agree that it could be more user friendly to propagate checked state down and I will try to fix it as soon as I have a little time.

More interesting is that you have files and folders in globalignore state. Could you provide some additional info given by clicking "Info" action in items context menu? Lets choice ".DS_Store" and "firefox-87.0" for example.

Which version of Syncthing you used?

**Comment by Jip-Hop** [comment]
Hi galilley,

My Syncthing version is v1.17.0. I'm using the [syncthing-macos](https://github.com/syncthing/syncthing-macos) Syncthing wrapper.

I re-applied the template, and removed all files in the folder. They still exist on the remote side obviously. Still many yellow items, even though info says they are not globally ignored. See the results below.  

<img width="433" alt="Screenshot 2021-07-08 at 14 54 48" src="https://user-images.githubusercontent.com/2871973/124925601-e912ad00-dffc-11eb-9e15-971bdbbb9ee6.png">

```json
Database:
{
    "availability": null,
    "global": {
        "deleted": false,
        "ignored": false,
        "invalid": false,
        "localFlags": 0,
        "modified": "2021-07-06T20:07:24.634152039+02:00",
        "modifiedBy": "ABCDEFG",
        "mustRescan": false,
        "name": "firefox-87.0.source/firefox-87.0/.DS_Store",
        "noPermissions": false,
        "numBlocks": 1,
        "permissions": "0644",
        "sequence": 780862,
        "size": 14340,
        "type": "FILE_INFO_TYPE_FILE",
        "version": [
            "ABCDEFG:1625594997"
        ]
    },
    "local": {
        "deleted": false,
        "ignored": true,
        "invalid": true,
        "localFlags": 2,
        "modified": "2021-07-06T20:07:24.634152039+02:00",
        "modifiedBy": "ABCDEFG",
        "mustRescan": false,
        "name": "firefox-87.0.source/firefox-87.0/.DS_Store",
        "noPermissions": false,
        "numBlocks": 0,
        "permissions": "0644",
        "sequence": 344187,
        "size": 0,
        "type": "FILE_INFO_TYPE_FILE",
        "version": [
            "ABCDEFG:1625594997"
        ]
    },
    "mtime": {
        "err": null,
        "value": {
            "real": "0001-01-01T00:00:00Z",
            "virtual": "0001-01-01T00:00:00Z"
        }
    }
}

Local:
{
    "size": 14340,
    "modified": "2021-07-06T20:07:24+02:00",
    "type": "FILE",
    "syncstate": "globalignore",
    "children": 0
}
```

<img width="427" alt="Screenshot 2021-07-08 at 14 56 24" src="https://user-images.githubusercontent.com/2871973/124925649-f3cd4200-dffc-11eb-9d60-eee6fcc5a2b2.png">


```json
Database:
{
    "availability": null,
    "global": {
        "deleted": false,
        "ignored": false,
        "invalid": false,
        "localFlags": 0,
        "modified": "2021-07-06T15:50:21.651308592+02:00",
        "modifiedBy": "ABCDEFG",
        "mustRescan": false,
        "name": "firefox-87.0.source/firefox-87.0/accessible",
        "noPermissions": false,
        "numBlocks": 0,
        "permissions": "0755",
        "sequence": 776088,
        "size": 128,
        "type": "FILE_INFO_TYPE_DIRECTORY",
        "version": [
            "ABCDEFG:1625580297"
        ]
    },
    "local": {
        "deleted": false,
        "ignored": true,
        "invalid": true,
        "localFlags": 2,
        "modified": "2021-07-06T15:50:21.651308592+02:00",
        "modifiedBy": "ABCDEFG",
        "mustRescan": false,
        "name": "firefox-87.0.source/firefox-87.0/accessible",
        "noPermissions": false,
        "numBlocks": 0,
        "permissions": "0755",
        "sequence": 338403,
        "size": 128,
        "type": "FILE_INFO_TYPE_DIRECTORY",
        "version": [
            "ABCDEFG:1625580297"
        ],
        "partial": false
    },
    "mtime": {
        "err": null,
        "value": {
            "real": "0001-01-01T00:00:00Z",
            "virtual": "0001-01-01T00:00:00Z"
        }
    }
}

Local:
{
    "size": 128,
    "modified": "2021-07-06T15:50:21+02:00",
    "type": "DIRECTORY",
    "syncstate": "globalignore",
    "children": 15
}
```

**Comment by Jip-Hop** [comment]
I quit syncthing-pyselective, opened it again, and now it seems to work:

<img width="446" alt="Screenshot 2021-07-08 at 15 08 15" src="https://user-images.githubusercontent.com/2871973/124927293-c08bb280-dffe-11eb-84a3-56512f534b5e.png">

No more globalignores. Strange 🤪

**Comment by galilley** [comment]
Dear Jip-Hop,
Thank you for the info!
Any way, I will check your case and try to find out the problem. Could you provide info for the same files again when everything seems to be ok?

Note: please keep in mind that syncthing-pyselective do not update info automatically, just only when you click "Get file tree".

**Comment by Jip-Hop** [comment]
For the same files as above:
```
Database:
{
    "availability": null,
    "global": {
        "deleted": false,
        "ignored": false,
        "invalid": false,
        "localFlags": 0,
        "modified": "2021-07-06T20:07:24.634152039+02:00",
        "modifiedBy": "ABCDEFG",
        "mustRescan": false,
        "name": "firefox-87.0.source/firefox-87.0/.DS_Store",
        "noPermissions": false,
        "numBlocks": 1,
        "permissions": "0644",
        "sequence": 780862,
        "size": 14340,
        "type": "FILE_INFO_TYPE_FILE",
        "version": [
            "ABCDEFG:1625594997"
        ]
    },
    "local": {
        "deleted": false,
        "ignored": true,
        "invalid": true,
        "localFlags": 2,
        "modified": "2021-07-06T20:07:24.634152039+02:00",
        "modifiedBy": "ABCDEFG",
        "mustRescan": false,
        "name": "firefox-87.0.source/firefox-87.0/.DS_Store",
        "noPermissions": false,
        "numBlocks": 0,
        "permissions": "0644",
        "sequence": 344187,
        "size": 0,
        "type": "FILE_INFO_TYPE_FILE",
        "version": [
            "ABCDEFG:1625594997"
        ]
    },
    "mtime": {
        "err": null,
        "value": {
            "real": "0001-01-01T00:00:00Z",
            "virtual": "0001-01-01T00:00:00Z"
        }
    }
}

Local:
{
    "size": 14340,
    "modified": "2021-07-06T20:07:24+02:00",
    "type": "FILE",
    "syncstate": "ignored",
    "children": 0
}
```

```
Database:
{
    "availability": null,
    "global": {
        "deleted": false,
        "ignored": false,
        "invalid": false,
        "localFlags": 0,
        "modified": "2021-07-06T15:50:21.651308592+02:00",
        "modifiedBy": "ABCDEFG",
        "mustRescan": false,
        "name": "firefox-87.0.source/firefox-87.0/accessible",
        "noPermissions": false,
        "numBlocks": 0,
        "permissions": "0755",
        "sequence": 776088,
        "size": 128,
        "type": "FILE_INFO_TYPE_DIRECTORY",
        "version": [
            "ABCDEFG:1625580297"
        ]
    },
    "local": {
        "deleted": false,
        "ignored": true,
        "invalid": true,
        "localFlags": 2,
        "modified": "2021-07-06T15:50:21.651308592+02:00",
        "modifiedBy": "ABCDEFG",
        "mustRescan": false,
        "name": "firefox-87.0.source/firefox-87.0/accessible",
        "noPermissions": false,
        "numBlocks": 0,
        "permissions": "0755",
        "sequence": 338403,
        "size": 128,
        "type": "FILE_INFO_TYPE_DIRECTORY",
        "version": [
            "ABCDEFG:1625580297"
        ],
        "partial": false
    },
    "mtime": {
        "err": null,
        "value": {
            "real": "0001-01-01T00:00:00Z",
            "virtual": "0001-01-01T00:00:00Z"
        }
    }
}

Local:
{
    "size": 128,
    "modified": "2021-07-06T15:50:21+02:00",
    "type": "DIRECTORY",
    "syncstate": "ignored",
    "children": 15
}
```

This is for a file that's selected:
```
Database:
{
    "availability": null,
    "global": {
        "deleted": false,
        "ignored": false,
        "invalid": false,
        "localFlags": 0,
        "modified": "2021-03-18T14:48:28+01:00",
        "modifiedBy": "ABCDEFG",
        "mustRescan": false,
        "name": "firefox-87.0.source/firefox-87.0/.cargo/config.in",
        "noPermissions": false,
        "numBlocks": 1,
        "permissions": "0644",
        "sequence": 471698,
        "size": 5728,
        "type": "FILE_INFO_TYPE_FILE",
        "version": [
            "ABCDEFG:1625569205"
        ]
    },
    "local": {
        "deleted": true,
        "ignored": false,
        "invalid": false,
        "localFlags": 0,
        "modified": "2021-07-08T15:04:45+02:00",
        "modifiedBy": "ABCDEFG",
        "mustRescan": false,
        "name": "firefox-87.0.source/firefox-87.0/.cargo/config.in",
        "noPermissions": false,
        "numBlocks": 0,
        "permissions": "0644",
        "sequence": 347746,
        "size": 0,
        "type": "FILE_INFO_TYPE_FILE",
        "version": []
    },
    "mtime": {
        "err": null,
        "value": {
            "real": "0001-01-01T00:00:00Z",
            "virtual": "0001-01-01T00:00:00Z"
        }
    }
}

Local:
{
    "size": 5728,
    "modified": "2021-03-18T14:48:28+01:00",
    "type": "FILE",
    "syncstate": "syncing",
    "children": 0
}
```

**Comment by galilley** [comment]
Thanks a lot! I will try to fix it as soon as I have a little time.

upd: the issue must be fixed in 0.2.2
